### PR TITLE
Docs: Update `formatterOptions` in `formatter` documentation

### PR DESCRIPTION
**Pull request by jaspreet57**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~~[ ] Added/Updated related documentation.~~
- ~~[ ] Added/Updated related tests.~~

## Short description of the change(s)
`formatterOptions` example is updated and some minor text fixes are also done.

Fix: #2820

### CODE REVIEW

File: packages/hint/docs/contributor-guide/how-to/formatter.md [hunk 1]
```diff
@@ -25,17 +25,19 @@ A `message` looks like this:
 
 ```json
 {
-    "column": "number", // The column number where the issue was found if applicable.
-    "line": "number", // The line number where the issue was found if applicable.
+    "location": {
+        "column": "number", // The column number where the issue was found if applicable.
+        "line": "number", // The line number where the issue was found if applicable.
+    },
     "message": "string", // The human friendly detail of the error.
```
Comment (jaspreet57) [inline packages/hint/docs/contributor-guide/how-to/formatter.md]: @sarvaje There are more properties in Problem type, should we include them here or not necessary for docs?
Actuall type is as follows:
```ts
/** A problem found by a hint  */
export type Problem = {
    /** The location number where the Problem is */
    location: ProblemLocation;
    /** A message providing more information about the Problem */
    message: string;
    /** The html element where the Problem is */
    sourceCode: string;
    /** The uri of the resource firing this event */
    resource: string;
    /** The name of the triggered hint */
    hintId: string;
    /** The category of the triggered hint */
    category: Category;
    /** The severity of the hint based on the actual configuration */
    severity: Severity;
    /** Indicate the language of the sourceCode */
    codeLanguage?: string;
};
```

Comment (sarvaje) [reply packages/hint/docs/contributor-guide/how-to/formatter.md]: IMO, we should include them.

Comment (jaspreet57) [reply packages/hint/docs/contributor-guide/how-to/formatter.md]: I have added other properties too. And I think it looks good.

File: packages/hint/docs/contributor-guide/how-to/formatter.md [hunk 2]
```diff
@@ -25,17 +25,24 @@ A `message` looks like this:
...
+    "sourceCode": "string", // The html element code where the issue was found if applicable.
```
Comment (sarvaje) [inline packages/hint/docs/contributor-guide/how-to/formatter.md]: This doesn't need to be the `html element code`.

what about `The piece of code where the issue was found if applicable`

Comment (jaspreet57) [reply packages/hint/docs/contributor-guide/how-to/formatter.md]: Updated

**Pull request merged**